### PR TITLE
Feature/sonar qube workflow integration

### LINKED ISSUE #60: SonarQube rules + workflow integration
PR Validation + precommit hook ?? <= research this

### PR DESCRIPTION
**Pull request by RamonKea-inh**

### CODE REVIEW

**Comment by RamonKea-inh** [comment]
The deployment only fails because the overall testing code coverage is to low

**Review by CyberAve2100 — DISMISSED** [review]
De workflows zien er erg netjes uit, goed bezig! 
Misschien kunnen we wat issues aan de board toevoegen om te zorgen dat deze tests wel een succes worden

File: .github/workflows/deploy-prod.yml
Comment (Tayam-El-Jebari) [inline .github/workflows/deploy-prod.yml:1]: why cache your SonarCloud packages and maven packages in pr-validation.yml, but not here?

File: .github/workflows/deploy-prod.yml [hunk 1]
```diff
@@ -18,12 +18,59 @@ jobs:
...
+          "${{ secrets.SONAR_HOST_URL }}/api/qualitygates/project_status?analysisId=$ANALYSIS_ID" | \
+          jq -r '.projectStatus.status')
+          
+          echo "Quality Gate Status: $QUALITY_GATE_STATUS"
+          
+          if [ "$QUALITY_GATE_STATUS" != "OK" ]; then
+            echo "❌ Quality Gate Failed"
+            exit 1
+          fi
+
```
Comment (Tayam-El-Jebari) [inline .github/workflows/deploy-prod.yml]: This works in theory, but when i searched online i found out that there exists an official github action from sonar themself which does the quality gate checking, see:
https://github.com/SonarSource/sonarqube-quality-gate-action?tab=readme-ov-file#usage
This has more error handling and better status reporting.

File: .github/workflows/deploy-dev.yml [hunk 2]
```diff
@@ -28,6 +30,54 @@ jobs:
...
+          ANALYSIS_ID=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?id=$TASK_ID" | \
+          jq -r '.task.analysisId')
+
+          # Check quality gate status
+          QUALITY_GATE_STATUS=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/qualitygates/project_status?analysisId=$ANALYSIS_ID" | \
+          jq -r '.projectStatus.status')
```
Comment (Tayam-El-Jebari) [inline .github/workflows/deploy-dev.yml]: same as comment below

File: .github/workflows/deploy-prod.yml [hunk 3]
```diff
@@ -18,12 +18,59 @@ jobs:
...
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?componentKey=${{ secrets.SONAR_PROJECT_KEY }}")
+          
+          TASK_ID=$(echo $SONAR_CE_TASK_URL | jq -r '.task.id')
+          
+          # Wait for task to complete and get analysis ID
+          ANALYSIS_ID=$(curl -s \
+          -u "${{ secrets.SONAR_TOKEN }}:" \
+          "${{ secrets.SONAR_HOST_URL }}/api/ce/task?id=$TASK_ID" | \
+          jq -r '.task.analysisId')
```
Comment (Tayam-El-Jebari) [inline .github/workflows/deploy-prod.yml]: same as comment below, there exists a github action for this.

**Comment by RamonKea-inh** [comment]
Code analyses are now done through Sonar Cloud completly out off the workflows at the cost off the code coverage quality gate, but this was to strict and not changeble in the free version so it might be an option for later.

**Review by SemPlaatsman — APPROVED** [review]

**Review by DaanTol — APPROVED** [review]

**Pull request merged**